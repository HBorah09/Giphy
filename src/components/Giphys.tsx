import { useState, FC } from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import Modal from './Modal';

export interface Gif {
    id: string,
    title: string,
    images: {
        original_still: {
            url: string,
            [propName: string]: any;
        },
        fixed_height: {
            url: string,
            [propName: string]: any;
        },
        [propName: string]: any;
    },
    [propName: string]: any;
}

interface Props {
    data: Gif[]
}

const Giphys: FC<Props> = ({ data }) => {

    const [showModal, setShowModal] = useState(false);
    const [modalData, setModalData] = useState({ id: '',
                                                  title: '',
                                                  images: {
                                                    original_still: { url: "" },
                                                    fixed_height: { url: "" }
                                                  }
                                                });

    const handleClose = () => setShowModal(false);
    
    const handleShow = (gif: Gif) => {
        setShowModal(true);
        setModalData(gif);
    }

    return(
        <>
        <Container>
            <Row>
                {data.map(gif => (
                    <Col key={`gif-${gif.id}`} xs={12} sm={4} md={4} lg={2} className="gif-container">
                        <img tabIndex={0} src={gif.images.original_still.url} alt={gif.title} className="gif" onClick={() => handleShow(gif)} />
                    </Col>
                ))}
            </Row>
        </Container>
        {showModal && <Modal show={showModal} handleClose={handleClose} data={modalData} />}
        </>
    )
}

export default Giphys;