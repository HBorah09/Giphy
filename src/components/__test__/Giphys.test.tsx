import { shallow } from 'enzyme'
import Giphys from '../Giphys';

describe('<Giphys />', () => {

    const props = {
        data: [
            {
                id: '123',
                title: 'funny gif',
                images: {
                    fixed_height: {
                        url: ''
                    }, 
                    original_still: {
                        url: ''
                    }
                },
            },
            {
                id: '456',
                title: 'funny gif',
                images: {
                    fixed_height: {
                        url: ''
                    }, 
                    original_still: {
                        url: ''
                    }
                },
            }
        ]
    }

    it('renders without crashing', () => {
        const wrapper = shallow(<Giphys { ...props } />);
        expect(wrapper).toMatchSnapshot();
    });
})