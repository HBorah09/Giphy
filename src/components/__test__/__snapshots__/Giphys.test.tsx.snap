// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Giphys /> renders without crashing 1`] = `
<Fragment>
  <Container
    fluid={false}
  >
    <Row>
      <Col
        className="gif-container"
        key="gif-123"
        lg={2}
        md={4}
        sm={4}
        xs={12}
      >
        <img
          alt="funny gif"
          className="gif"
          onClick={[Function]}
          src=""
          tabIndex={0}
        />
      </Col>
      <Col
        className="gif-container"
        key="gif-456"
        lg={2}
        md={4}
        sm={4}
        xs={12}
      >
        <img
          alt="funny gif"
          className="gif"
          onClick={[Function]}
          src=""
          tabIndex={0}
        />
      </Col>
    </Row>
  </Container>
</Fragment>
`;
